src/main/java/it/pagopa/pm/gateway/service/ClientsConfigService.java

```diff
@@ -0,0 +1,35 @@
+package it.pagopa.pm.gateway.service;
+
+import com.fasterxml.jackson.core.JsonProcessingException;
+import com.fasterxml.jackson.core.type.TypeReference;
+import com.fasterxml.jackson.databind.ObjectMapper;
+import it.pagopa.pm.gateway.dto.ClientConfig;
+import org.springframework.beans.factory.annotation.Value;
+import org.springframework.context.annotation.Scope;
+import org.springframework.stereotype.Service;
+
+import javax.annotation.PostConstruct;
+import java.util.HashMap;
+import java.util.Map;
+
+@Service
+@Scope("singleton")
+public class ClientsConfigService {
+    @Value("${pgs.clients.config}")
+    private String clientsConfigJson;
+    private Map<String, ClientConfig> clientsConfigMap;
+
+    @PostConstruct
+    private void convertJsonToMap() throws JsonProcessingException {
+        TypeReference<HashMap<String, ClientConfig>> mapType = new TypeReference<HashMap<String, ClientConfig>>() {};
+        clientsConfigMap = new ObjectMapper().readValue(clientsConfigJson, mapType);
+    }
+
+    public ClientConfig getByKey(String key) {
+        return clientsConfigMap.get(key);
+    }
+
+    public boolean containsKey(String key) {
+        return clientsConfigMap.containsKey(key);
+    }
+}
```

Comment:
* Semantically it's not really a service (it doesn't provide an API to do side-effects or business logic)
* Rename `clientsConfigMap` to `clientsConfig` or `config`, hungarian notation is unnecessary here
* The config's type is `Map<String, ClientConfig>`, do we need to accept any string as a configuration key?
* Prefer constructor injection to make tests easier
```java
public ClientsConfigService(@Autowired ObjectMapper mapper, @Value("${pgs.clients.config}") String clientsConfig) throws JsonProcessingException {
        TypeReference<HashMap<String, ClientConfig>> mapType = new TypeReference<HashMap<String, ClientConfig>>() {};
        clientsConfigMap = Collections.unmodifiableMap(mapper.readValue(clientsConfig, mapType));
}
```
With this you don't even need to store the string containing the serialized configuration. 

Moreover, with an unmodifiable collection initialized during construction you could even think of making it a `public final` member of this class (if you think that enumerating configuration keys is an operation you're planning to use it would be a better design rather than copying `java.util.Map`'s API inside this class). (As far as testing goes you could make a protected constructor taking an initialized map in the constructor to make everything easier).

Also, you could make this class a POJO and write a separate [converter](https://www.baeldung.com/spring-type-conversions) so that in principle you could write:
```java
class Foo {
    public Foo(@Value("pgs.clients.config") ClientsConfigService config) {
        // --- snip ---
    }
}
```

Reply:
For the third point: I decided not to use an Enum because if we'll need to add a new client configuration in the future, we'll be able to do it by modifying the env value only without the need to change the code. 

For the `java.util.Map`'s APIs copied inside the methods: I didn't really like the idea of doing `clientsConfigService.getConfig().get(key)`, but if you prefer it this way I can make the parameter `public final`. Let me know.

For the converter: it's a nice idea but I'm not sure it's worth it since we need to make the conversion only once.

Reply:
https://github.com/pagopa/pagopa-payment-transactions-gateway/pull/196/commits/9a8a7becba2a16f4762c739cd3fd79cfc9f03c4b